src/main/java/jenkins/plugins/office365connector/CardBuilder.java

```diff
@@ -194,41 +192,37 @@ public Card createBuildMessageCard(StepParameters stepParameters) {
     }
 
     private void addScmDetails() {
-        Set<User> users;
-        List<ChangeLogSet<ChangeLogSet.Entry>> sets;
-
-        try {
-            users = (Set<User>) run.getClass().getMethod("getCulprits").invoke(run);
-            sets = (List<ChangeLogSet<ChangeLogSet.Entry>>) run.getClass().getMethod("getChangeSets").invoke(run);
-        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-            users = Collections.emptySet();
-            sets = Collections.emptyList();
+        if (!(run instanceof AbstractBuild)) {
+            return;
         }
+        AbstractBuild abstractBuild = (AbstractBuild) run;
+        factsBuilder.addCulprits(abstractBuild.getCulprits());
 
-        factsBuilder.addCulprits(users);
-
-        if (!sets.isEmpty()) {
-            Set<User> authors = new HashSet<>();
-            int filesCounter = 0;
-            if (Iterables.all(sets, Predicates.instanceOf(ChangeLogSet.class))) {
-                for (ChangeLogSet<ChangeLogSet.Entry> set : sets) {
-                    for (ChangeLogSet.Entry entry : set) {
-                        authors.add(entry.getAuthor());
-                        filesCounter += getAffectedFiles(entry).size();
-                    }
+        List<ChangeLogSet<ChangeLogSet.Entry>> sets = abstractBuild.getChangeSets();
+        if (sets.isEmpty()) {
+            return;
+        }
+        Set<User> authors = new HashSet<>();
+        int filesCounter = 0;
+        if (Iterables.all(sets, Predicates.instanceOf(ChangeLogSet.class))) {
```

Comment:
This actually seems unnecessary consider using Java stream instead of iterables at least.

Reply:
could be nicer but I need to modify two variables inside the lambda expression which makes the code harder to read
will leave for now, maybe later will find nicer idea for this :-)